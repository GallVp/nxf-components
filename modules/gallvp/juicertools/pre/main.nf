process JUICERTOOLS_PRE {
    tag "$meta.id"
    label 'process_medium'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/juicertools:2.20.00--hdfd78af_0':
        'biocontainers/juicertools:2.20.00--hdfd78af_0' }"

    input:
    tuple val(meta), path(txt)
    path(index)
    path(sizes)

    output:
    tuple val(meta), path("*.hic"), emit: hic
    path "versions.yml"           , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    def index_arg = index ? "-i $index" : ''
    def avail_mem = (task.memory.giga*0.8).intValue()
    """
    mkdir user_home
    export _JAVA_OPTIONS="-Djava.util.prefs.userRoot=user_prefs -Duser.home=user_home -Xms${avail_mem}g -Xmx${avail_mem}g"

    juicer_tools \\
        pre \\
        $args \\
        --threads $task.cpus \\
        $index_arg \\
        $txt \\
        ${prefix}.hic \\
        $sizes


    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        juicertools: \$(juicer_tools --version |& sed -n 's|Juicer Tools Version\\(.*\\)|\\1|p')
    END_VERSIONS
    """

    stub:
    def prefix = task.ext.prefix ?: "${meta.id}"
    """

    touch ${prefix}.hic

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        juicertools: \$(juicer_tools --version |& sed -n 's|Juicer Tools Version\\(.*\\)|\\1|p')
    END_VERSIONS
    """
}
