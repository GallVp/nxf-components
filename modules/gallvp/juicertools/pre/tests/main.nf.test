nextflow_process {

    name "Test Process JUICERTOOLS_PRE"
    script "../main.nf"
    config "./nextflow.config"
    process "JUICERTOOLS_PRE"

    tag "modules"
    tag "modules_gallvp"
    tag "juicertools"
    tag "juicertools/pre"
    tag "yahs/juicerpre"
    tag "sort"

    setup {
        run("YAHS_JUICERPRE") {
            script "../../../yahs/juicerpre/main.nf"
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file( 'https://github.com/Plant-Food-Research-Open/assemblyqc/raw/4f9fba392cf7212d61c6245fd72339e09164ce44/tests/hic/testdata/HYh1h2_Chr01_6000000_6015000.bam' , checkIfExists: true)
                ]

                input[1] = file( 'https://github.com/Plant-Food-Research-Open/assemblyqc/raw/4f9fba392cf7212d61c6245fd72339e09164ce44/tests/hic/testdata/HYh1h2_Chr01_6000000_6015000.agp' , checkIfExists: true)
                input[2] = file( 'https://github.com/Plant-Food-Research-Open/assemblyqc/raw/4f9fba392cf7212d61c6245fd72339e09164ce44/tests/hic/testdata/HYh1h2_Chr01_6000000_6015000.fasta.fai' , checkIfExists: true)
                """
            }
        }

        run("SORT") {
            script "../../../sort/main.nf"

            process {
                """
                input[0] = YAHS_JUICERPRE.out.txt
                """
            }
        }
    }

    test("hongyang - hic") {

        when {
            process {
                """
                input[0] = SORT.out.sorted

                def sizes = file( 'https://github.com/Plant-Food-Research-Open/assemblyqc/raw/4f9fba392cf7212d61c6245fd72339e09164ce44/tests/hic/testdata/HYh1h2_Chr01_6000000_6015000.fasta.fai' , checkIfExists: true)
                def sizesFile = new File('hongyang.sizes')
                sizesFile.text = sizes.text.split('\\n').collect { it.split('\\t')[0..1].join('\\t') }.join('\\n')

                input[1] = sizesFile.toPath()
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("hongyang - hic - stub") {

        options '-stub'

        when {
            process {
                """
                input[0] = SORT.out.sorted

                def sizes = file( 'https://github.com/Plant-Food-Research-Open/assemblyqc/raw/4f9fba392cf7212d61c6245fd72339e09164ce44/tests/hic/testdata/HYh1h2_Chr01_6000000_6015000.fasta.fai' , checkIfExists: true)
                def sizesFile = new File('hongyang.sizes')
                sizesFile.text = sizes.text.split('\\n').collect { it.split('\\t')[0..1].join('\\t') }.join('\\n')

                input[1] = sizesFile.toPath()
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}