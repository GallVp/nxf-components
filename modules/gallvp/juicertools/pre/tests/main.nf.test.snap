{
    "hongyang - hic - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.hic:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,2f2429fdc6605a1e22325baff225434f"
                ],
                "hic": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.hic:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,2f2429fdc6605a1e22325baff225434f"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.6"
        },
        "timestamp": "2025-05-23T15:12:04.699874"
    },
    "hongyang - hic - mode - assembly - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.hic:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,2f2429fdc6605a1e22325baff225434f"
                ],
                "hic": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.hic:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,2f2429fdc6605a1e22325baff225434f"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-08-07T16:20:15.160418"
    },
    "hongyang - hic": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.hic:md5,5ce7a683ccbba1556fb825cd87163dee"
                    ]
                ],
                "1": [
                    "versions.yml:md5,2f2429fdc6605a1e22325baff225434f"
                ],
                "hic": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.hic:md5,5ce7a683ccbba1556fb825cd87163dee"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,2f2429fdc6605a1e22325baff225434f"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.6"
        },
        "timestamp": "2025-05-23T15:11:55.622958"
    }
}
